### PR TITLE
Fix to handling of column names with special chars

### LINKED ISSUE #22: aov_car does not cope with column names with spaces in them (or other special chars)
```
data <- list("dependent" = rnorm(100), "RM Factor 1" = factor(rep(c("Level 1", "Level 2"), 50)), "subject" = factor(rep(1:50, each = 2)))

attr(data, 'row.names') <- seq_len(length(data[[1]]))
attr(data, 'class') <- 'data.frame'

formula <- as.formula("dependent ~ `RM Factor 1` + Error(subject/(`RM Factor 1`))")

afex::aov_car(formula, data)
```

### PR DESCRIPTION
**Pull request by jonathon-love**
fixes #22

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
+
+escape_vars <- function(names) {
+  if (length(names) == 0)
+    return(names)
+  names <- sapply(names, function(name) {
```
Comment (singmann) [inline R/utils.R]: can you please change `sapply` to `vapply`? I had too many problems with the former in `MPTinR` as it does not guarantee a specific return value. So I only use `sapply` in interactive use, but do not want it in package code.

**Comment by singmann** [comment]
hey, that looks all pretty good. but what is the logic in the `decomposeTerm` function. I cannot immediately grasp it fully? What is going on with the backslahses? I would like to make sure I understand what it is supposed to do.

And I can only accept the push if you are willing to change the `sapply` to `vapply`.

**Comment by jonathon-love** [comment]
decomposeTerm() splits a term into it's components:
```
fred:jim:bob -> "fred" "jim" "bob"
```
this can include backsticks:
```
`fred mcbob`:jim:bob -> "fred mcbob" "jim" "bob"
```
and backsticks can be escaped:
```
`fred mc\`bob`:jim:bob -> "fred mc`bob" "jim" "bob"
```
but you can't just look for (and ignore):
```
\`
```
because it could be like this:
```
`fred mcbob\\`:jim:bob -> "fred mcbob\" "jim" "bob"
```
so that's why there's that stuff with the backslashes.

makes sense?

i'll update the PR to user vapply instead.

cheers

**Comment by singmann** [comment]
Thanks. And sorry again for being super slow.

**Pull request merged**